### PR TITLE
Add E2E integration test pipeline

### LINKED ISSUE #807: Functions Integration Tests—Onboard Python worker
Please follow the instructions in the OneNote to onboard your language worker.

### PR DESCRIPTION
**Pull request by Hazhzeng**
### Description
This test pipeline is a part of story to finish the whole daily integration platform.
1. Each commit in Python Worker, the original azure-pipelines.yml now generates a core tools with the latest worker from dev branch
2. The Azure DevOps pipeline will now generates the artifact https://azfunc.visualstudio.com/Azure%20Functions/_packaging?_a=feed&feed=AzureFunctionsPreRelease
3. The daily run .c1/e2e_integration_test/pipeline.yml will pick up the core tools artifact from above, and invoke against tests/endtoend Python worker E2E tests.

### Changes
1. Remove unused **.ci/e2e** pipeline
2. Split the `python setup.py webhost` task into two different tasks
    a. python setup.py webhost => compile webjobs host from latest release source code
    b. python setup.py extension => restore the webjobs trigger extension defined in setup.py::AZURE_EXTENSIONS
3. Add a new pipeline in Azure DevOps [Azure.azure-functions-python-worker.e2e-integration-tests](https://azfunc.visualstudio.com/Azure%20Functions/_build?definitionId=59&_a=summary&view=runs)

### Next Step
Use the integration orchestrator pipeline to call this E2E test pipeline

Fixes #807

/cc: @Francisco-Gamino 

---

### PR information

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azure_functions_worker/constants.py [hunk 1]
```diff
@@ -11,8 +11,10 @@
 RPC_HTTP_TRIGGER_METADATA_REMOVED = "RpcHttpTriggerMetadataRemoved"
 WORKER_STATUS = "WorkerStatus"
 
-# Debug Flags
+# Debug Flags and Worker Test Util Configuration
 PYAZURE_WEBHOST_DEBUG = "PYAZURE_WEBHOST_DEBUG"
+PYAZURE_WORKER_DIR = "PYAZURE_WORKER_DIR"
```
Comment (vrdmr) [inline azure_functions_worker/constants.py:16]: Do you envision a new flag coming in for Test specific configurations in the future? If yes, then please split this group of variables into two.

Comment (vrdmr) [inline azure_functions_worker/constants.py:16]: Do you envision a new flag coming in for Test specific configurations in the future? If yes, then please split this group of variables into two.

**Review by vrdmr — APPROVED** [review]
Thanks a lot for the change. LGTM.

@Francisco-Gamino Could you please take a look?

File: .ci/e2e_integration_test/start-e2e.ps1 [hunk 2]
```diff
@@ -0,0 +1,95 @@
+#
```
Comment (Francisco-Gamino) [inline .ci/e2e_integration_test/start-e2e.ps1:1]: Please add a note at the top of the file that explains that we are running the Python language worker E2E tests against a prerelease version of the Core Tools (from Integration Tests).

**Review by Francisco-Gamino — APPROVED** [review]
Two minor comments. Other than that, LGTM.

These are the files I reviewed:
.ci/e2e_integration_test/start-e2e.ps1
.ci/e2e_integration_test/pipeline.yml

File: azure_functions_worker/constants.py (hunk 1 above)
Comment (Hazhzeng) [reply azure_functions_worker/constants.py:16]: Nice catch! Yes, this is a new setting for the integration test specifically, will split this into two.

**Pull request merged**